src/packages/api/api_manager.py

```diff
@@ -0,0 +1,468 @@
+            rescue (Rescue): The Rescue object to create in the API.
+        """
+        data = APIManager._serialize_rescue(rescue)
+        del data['id']
+        if data['relationships']['firstLimpet']['data'] is None:
+            del data['relationships']['firstLimpet']
+        jsonapi_data = {'data': data}
+        new_rescue = await self._call('rescues', method='POST', body=jsonapi_data)
+        breakpoint()
+        return APIManager._deserialize_rescue(new_rescue)
+
+    async def update_rescue(self, rescue: Rescue):
+        """
+        Update the case in the API with the data from new_data.
+
+        Args:
+            rescue (Rescue): The Rescue object to update in the API.
+        """
+        data = APIManager._serialize_rescue(rescue)
+        jsonapi_data = {'data': data}
+        await self._call(f"rescues/{rescue.api_id}", method='PATCH', body=jsonapi_data)
+
+    # @TODO: @functools.lru_cache()
+    async def get_rat(self, uuid: str):
+        """
+        Use the API to get a rat by UUID.
+
+        Args:
+            uuid (str): The Rat's UUID.
+
+        Returns:
+            A RatData object describing the Rat found, or None if none was found.
+        """
+        if not uuid:
+            raise ValueError("uuid must be specified!")
+
+        try:
+            data = await self._call(f"rats/{uuid}")
+            return APIManager._deserialize_rat(data['data'])
+        except aiohttp.ClientError:
+            return None
+
+    async def find_rat(self,
+                       name: str,
+                       platform: typing.Optional[Platforms] = None) -> typing.Optional[Rat]:
+        """
+        Use the API to find a rat by name, and optionally platform.
+
+        Args:
+            name (str): The Rat's name.
+            platform (Platforms): The Rat's platform.
+
+        Returns:
+            A RatData object describing the Rat found, or None if none was found.
+        """
+        if not name:
+            raise ValueError("name must be specified!")
+
+        params = {'filter[name:ilike]': name}
+        if platform:
+            params['filter[platform:eq]'] = platform.name.casefold()
+
+        try:
+            data = await self._call('rats', params)
+            if data:
+                return APIManager._deserialize_rat(data['data'][0])
+        except aiohttp.ClientError:
+            return None
+
+    @staticmethod
+    def _build_uri_params(params: typing.Optional[typing.Dict[str, str]]) -> str:
+        """
+        Build a URI query string out of a Dict of key/value pairs.
+
+        Args:
+            params (Dict): The key/value pairs to build into a query.
+
+        Returns:
+            A properly HTML-escaped URI query string with leading question mark.
+        """
+        param_string = ""
+        if params:
+            param_string = '&'.join(
+                [f"{key}={escape(str(value))}" for key, value in params.items()]
+            )
+        return f"?{param_string}"
```

Comment:
is there anything in our dependencies we can leverage (such as `requests`) to allow us not to need implement a method like this ourselves?